### PR TITLE
Merge Zcheri_mode and Zcheri_legacy

### PR DESCRIPTION
**Pull request by arichardson**
The main reason for a separate Zcheri_mode extensions is that it freed
up a bit from the 32-bit capability format. With a new 32-bit
permissions encoding (https://github.com/riscv/riscv-cheri/pull/155), we
need less than a full bit for the mode since it is only required for
executable capabilities rather than fully orthogonal. Removing the
separate Zcheri_mode extension simplifies the architecture and will
allow for further simplification of the CME architectural state.

Co-authored-by: Tariq Kurd <tariq.kurd@codasip.com>
Co-authored-by: Andres Amaya Garcia <andres.amaya@codasip.com>

### CODE REVIEW

**Comment by arichardson** [comment]
CC: @sorear

File: src/riscv-mode-integration.adoc [hunk 1]
```diff
@@ -1,14 +1,7 @@
...
-
-{cheri_mode_ext_name} is an optional extension to {cheri_legacy_ext_name}.
-Implementations that support {cheri_mode_ext_name} allow fine-grained switching
-between Capability and Legacy modes using indirect jump instructions.
-
 [#section-cheri-execution-mode]
-=== CHERI Execution Mode
+=== Capability format
 
-{cheri_mode_ext_name} adds a new CHERI execution mode bit (M) to capabilities.
```
Comment (arichardson) [inline src/riscv-mode-integration.adoc:11]: We can probably fold this file into riscv-legacy-integration.adoc, but I wanted to keep the changes smaller initially.

File: src/riscv-legacy-integration.adoc [hunk 2]
```diff
@@ -93,8 +92,7 @@ the sign-extended offset.
 The <<JALR_MODE>> instruction is modal, giving access to the functionality of
 <<JALR>> from either operating mode.
 
-NOTE: <<JALR_MODE>> can be used to change the current CHERI execution mode when
-the implementation supports {cheri_mode_ext_name}.
+NOTE: <<JALR_MODE>> can be used to change the current CHERI execution mode.
```
Comment (tariqkurd-repo) [inline src/riscv-legacy-integration.adoc:95]: not any more - and I merged the jalr.mode PR so I'm not sure why this section is still here

**Review by andresag01 — CHANGES_REQUESTED** [review]
Thanks for the changes. It looks good, just a few minor comments. It would be good to rebase this on top of the latest main which has the CME bits removal.

**Comment by andresag01** [comment]
@tariqkurd-repo @arichardson : I cherry picked the commit removing the CRE bit and re-arranged the Zcheri_legacy chapter a little. Here are my changes in case they are helpful: https://github.com/andresag01/riscv-cheri/commits/users/arichardson/spr/merge-zcheri_mode-and-zcheri_legacy/

File: src/riscv-legacy-integration.adoc [hunk 3]
```diff
@@ -65,9 +65,8 @@ top is 2^XLENMAX^
 [#section_legacy_ext_instructions]
```
[the remark this answers is not in the document]
Comment (arichardson) [reply src/riscv-legacy-integration.adoc:78]: Thanks for your changes, this should be resolved now.

**Review by andresag01 — APPROVED** [review]

**Pull request merged**